styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -219,31 +217,32 @@ private WorkflowConfiguration readFromJsonWithDefaults(ByteString payload)
     return WorkflowConfigurationBuilder.from(workflowConfig).deploymentTime(time.get()).build();
   }
 
-  private Response<Collection<WorkflowWithState>> workflowsWithState() {
-    try {
-      Collection<WorkflowWithState> workflows = storage.workflowsWithState().values();
-      return Response.forPayload(workflows);
-    } catch (IOException e) {
-      throw new RuntimeException("Failed to get workflows", e);
-    }
-  }
-
-  private Response<Collection<Workflow>> workflows(Request request) {
+  private Response<? extends Collection<?>> workflows(Request request) {
     try {
       var paramFilters = Stream.of(QueryParams.values())
           .map(e -> getFilterParams(request, e).map(param -> Map.entry(e, param)))
           .flatMap(Optional::stream)
           .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
 
-      Collection<Workflow> workflows = storage.workflows().values();
+      Collection<WorkflowWithState> workflowsWithState = storage.workflowsWithState().values();
+      Collection<WorkflowWithState> filteredWorkflows = filterWorkflows(workflowsWithState, paramFilters);
+      Optional<String> includeStates = getFilterParams(request, INCLUDE_STATES);
 
-      return Response.forPayload(filterWorkflows(workflows, paramFilters));
+      if (shouldIncludeState(includeStates)) {
+        return Response.forPayload(filteredWorkflows);
+      }
 
+      List<Workflow> workflows = filteredWorkflows.stream().map(WorkflowWithState::workflow).collect(Collectors.toList());
+      return Response.forPayload(workflows);
     } catch (IOException e) {
       throw new RuntimeException("Failed to get workflows", e);
     }
   }
 
+  private boolean shouldIncludeState(Optional<String> includeStates) {
+    return includeStates.filter(Boolean::parseBoolean).isPresent();
```

Comment:
it is not sufficient to just have ?full, I added a test to show it here https://github.com/spotify/styx/pull/1065/files#diff-a92738177ebe75548687fdc45ffa62a2b696244ac699584214fa96cb7e14cf35R823

Reply:
It currently does not work because of `Boolean::parseBoolean` I guess. But we could make it work, and have a value-less query parameter in this case seems to be more reasonable than having `full=true` or `full=false`.